akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -95,8 +95,10 @@ import scala.concurrent.{ Future, Promise }
             if (unbindStarted) {
               unbindCompleted()
             } else {
-              failStage(new IllegalStateException("IO Listener actor terminated unexpectedly for remote endpoint [" +
-                endpoint.getHostString + ":" + endpoint.getPort + "]"))
+              val ex = new IllegalStateException("IO Listener actor terminated unexpectedly for remote endpoint [" +
+                endpoint.getHostString + ":" + endpoint.getPort + "]")
+              unbindPromise.failure(ex)
```

Comment:
do we need to try to complete those in `postStop` also?

Reply:
Good point, yes in case of abrupt termination.

Reply:
could it have been completed already? there is a `unbindPromise.trySuccess(Done)` followed by `scheduleOnce` below

Reply:
Unsure, but better safe than sorry. I'll make it a tryFailure